include/cocaine/rpc/asio/header_definitions.hpp

```diff
@@ -0,0 +1,874 @@
+    You should have received a copy of the GNU Lesser General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef HEADER_DEFINITIONS_HPP
+#define HEADER_DEFINITIONS_HPP
+
+namespace cocaine { namespace io {
+struct headers {
+    friend class header_static_table_t;
+
+private:
+    struct detail {
+        struct empty_placeholder {
+            static
+            constexpr
+            header::data_t
+            name() {
+                return header::create_data("");
+            }
+
+            static
+            constexpr
+            header::data_t
+            value() {
+                return header::create_data("");
+            }
+        };
+        template <class Value>
+        struct value_mixin {
+            static
+            constexpr
+            header::data_t
+            value() {
+                return Value::value();
+            }
+        };
+    };
+
+public:
+    template<class Header>
+    static
+    header_t
+    make_header() {
+        return header_t(Header::name(), Header::value());
+    }
+
+    struct default_values_t {
+        struct zero_uint_value_t {
+            static
+            constexpr
+            header::data_t
+            value() {
+                return header::create_data("\0\0\0\0\0\0\0\0");
+            }
+        };
+
+        struct empty_string_value_t {
+            static
+            constexpr
+            header::data_t
+            value() {
+                return header::create_data("");
+            }
+        };
+
+        struct get_value_t {
+            static
+            constexpr
+            header::data_t
+            value() {
+                return header::create_data("GET");
+            }
+        };
+
+        struct post_value_t {
+            static
+            constexpr
+            header::data_t
+            value() {
+                return header::create_data("POST");
+            }
+        };
+
+        struct root_value_t {
+            static
+            constexpr
+            header::data_t
+            value() {
+                return header::create_data("/");
+            }
+        };
+
+        struct index_value_t {
+            static
+            constexpr
+            header::data_t
+            value() {
+                return header::create_data("/index.html");
+            }
+        };
+
+        struct http_value_t {
+            static
+            constexpr
+            header::data_t
+            value() {
+                return header::create_data("http");
+            }
+        };
+
+        struct https_value_t {
+            static
+            constexpr
+            header::data_t
+            value() {
+                return header::create_data("https");
+            }
+        };
+
+        struct status_200_value_t {
+            static
+            constexpr
+            header::data_t
+            value() {
+                return header::create_data("200");
```

Comment:
Has not convinced.

Check any robust HTTP2 implementation for any language (C++ for example).